pkg/osquery/table/kolide_target_membership.go

```diff
@@ -0,0 +1,123 @@
+package table
+
+import (
+	"context"
+	"time"
+
+	"github.com/boltdb/bolt"
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/gogo/protobuf/proto"
+	"github.com/kolide/launcher/pkg/osquery"
+	"github.com/kolide/launcher/pkg/pb/kt"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+const targetMembershipKey = "target_membership"
+
+type TargetMembership struct {
+	logger       log.Logger
+	db           *bolt.DB
+	targetClient kt.KTargetClient
+}
+
+func NewTargetMembership(logger log.Logger, db *bolt.DB, targetClient kt.KTargetClient) TargetMembership {
+	return TargetMembership{
+		logger:       logger,
+		db:           db,
+		targetClient: targetClient,
+	}
+}
+
+func (tm *TargetMembership) Plugin() *table.Plugin {
+	columns := []table.ColumnDefinition{
+		table.TextColumn("id"),
+	}
+
+	return table.NewPlugin("kolide_target_membership", columns, tm.generate())
+}
+
+func (tm *TargetMembership) generate() table.GenerateFunc {
+	return func(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+
+		targets, err := tm.GetTargetMembership(ctx)
+		if err != nil {
+			return nil, errors.Wrap(err, "getting target membership")
+		}
+
+		results := make([]map[string]string, len(targets))
+		for i, t := range targets {
+			results[i] = map[string]string{"id": t.GetId()}
+		}
+
+		return results, nil
+	}
+}
+
+func (tm *TargetMembership) Run(ctx context.Context) error {
```

Comment:
It's a little odd to me that this is in `pkg/osquery/table`, even on the same struct as the table... IMO `Run`, `GetTargetMembership`, and `UpdateTargetMembership` should be on their own struct. If necessary, an instance of that type can be a member of this struct.

Reply:
That seems unlike the other examples here. Certainly `GetTargetMembership` feels like it be on the `TargetMembership` struct. 

I dunno if we should make another struct for the fetches. Maybe depends where we want to go.

Reply:
i've refactored the Plugin and generate methods to be funcs instead, which just take a db handle and are no longer part of this type